### DISCUSSION TITLE
Dose Caddy DNS challenge needs Email Address？

### DISCUSSION THREAD

**Discussion by toyo2333** [SamEdwardes/personal-blog#28] [General]
hi，i just read your blog ：https://samedwardes.com/2023/11/19/homelab-tls-with-caddy-and-cloudflare/

what i confused is some other article said it needs a email （cloudflare email address）：

https://emby.media/community/index.php?/topic/102088-emby-caddy-v2-setup-with-cloudflare/
https://github.com/SlothCroissant/caddy-cloudflaredns

so what is the difference? which one is the correct way to config it?

thanks

Ray

**Comment by SamEdwardes** [comment]
Hi Ray,


Thank you for the question, and I apologize for the delay. In my experience, you do NOT need an email address. The only thing that Caddy needs to "talk" to Cloudflare is the API token. My example Caddyfile will work assuming you have `CLOUDFLARE_API_TOKEN` set as an environment variable.

```Caddyfile
test.edwardes.xyz {
    # Set this path to your site's directory.
    root * /usr/share/caddy

    # Enable the static file server.
    file_server

    tls {
        dns cloudflare {env.CLOUDFLARE_API_TOKEN}
    }
}
```

Does this work for you?
